### PR TITLE
chore: type improvement 6

### PR DESCRIPTION
**Pull request by nicolasburtey**
@dolcalmi if you want some Typescript types challenges: open this PR and see if you can remove my typescript directives and get the correct types :) 

this is the last PR that would activate no-implicity-any in the main codebase

### CODE REVIEW

**Comment by samerbuna** [comment]
@dolcalmi @vindard can you please review this? I think it's in a good stage to merge (there still a lot more to do, but let's get the check in first)

File: src/graphql/main/queries.ts [hunk 1]
```diff
@@ -30,24 +32,32 @@ const fields = {
...
+            [ACCOUNT_USERNAME]: domainAccount?.username,
+            [SemanticAttributes.HTTP_CLIENT_IP]: ip,
+          },
+          () => original(source, args, context, info),
+        )
+      }
     }
   }
-  return fields
+  return trcFields
```
Comment (nicolasburtey) [inline src/graphql/main/queries.ts:60]: what does `tcr` stands for?

File: src/graphql/root/query/on-chain-tx-fee-query.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ const OnChainTxFeeQuery = GT.Field({
 
     const fee = await Wallets.getOnChainFee({
       walletId,
-      account: domainAccount,
+      account: domainAccount as Account,
```
Comment (nicolasburtey) [inline src/graphql/root/query/on-chain-tx-fee-query.ts:31]: why not having `{ domainAccount: Account }` at the function signature above? we're looking some type guarantee with the casting here

File: src/graphql/main/queries.ts (hunk 1 above)
Comment (samerbuna) [reply src/graphql/main/queries.ts:60]: trace

File: src/graphql/root/query/on-chain-tx-fee-query.ts (hunk 2 above)
Comment (samerbuna) [reply src/graphql/root/query/on-chain-tx-fee-query.ts:31]: That type should come from the context, I plan to add the context type to all resolvers in the next PR.

File: src/servers/middlewares/wallet-id.ts [hunk 3]
```diff
@@ -1,29 +1,54 @@
...
   if (walletId instanceof Error) return walletId
 
-  const hasPermissions = await Accounts.hasPermissions(context.domainUser.id, walletId)
+  if (!context.domainUser) {
+    return new Error("Invalid wallet")
+  }
+
+  const hasPermissions = await Accounts.hasPermissions(
+    context.domainUser.id,
+    walletId as WalletId,
```
Comment (vindard) [inline src/servers/middlewares/wallet-id.ts:25]: Just a note-to-self that this type is properly applied in the `scalar/wallet-id.ts` scalar object and that this casting will likely be removed later on once the args are properly types here

**Review by vindard — COMMENTED** [review]
Has conflicts that need resolving, but otherwise looks good to me

File: src/graphql/root/query/on-chain-tx-fee-query.ts (hunk 2 above)
Comment (nicolasburtey) [reply src/graphql/root/query/on-chain-tx-fee-query.ts:31]: agree but I may be missing something: the casting is still unnecessary here compare to the existing code?

Comment (samerbuna) [reply src/graphql/root/query/on-chain-tx-fee-query.ts:31]: We have 2 context types (one for public and one for authenticated). The default is public, so the type here is `Account | undefined`.  This is an authenticated query, so we either cast here or use the authenticated context type. I faced some issues using the authenticated context type, so I'll probably come up with a different option.

Comment (nicolasburtey) [reply src/graphql/root/query/on-chain-tx-fee-query.ts:31]: my point is that the line above was already making `Account` a not undefined field?

https://github.com/GaloyMoney/galoy/pull/1384/files#diff-4fed0ac3ed1e683af6718bfe6cf2825f4eacedadc4dab1be89b387e8b29b8b6dR19

**Review by nicolasburtey — COMMENTED** [review]
still have a question on https://github.com/GaloyMoney/galoy/pull/1384/files#r926395160 but going to merge it so we start to get more type enforcement in the core codebase

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**